### PR TITLE
assets-tx with configured asset

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

sattelite changes according parity process (same as here https://github.com/paritytech/cumulus/pull/1572) for changes in https://github.com/paritytech/substrate/pull/12087 (and our PR https://github.com/ComposableFi/substrate/pull/12)

### CODE REVIEW

File: parachains/runtimes/assets/statemint/src/lib.rs [hunk 1]
```diff
@@ -531,12 +531,23 @@ impl pallet_collator_selection::Config for Runtime {
...
+	pub const UseUserConfiguration: bool = false;
+}
+
 impl pallet_asset_tx_payment::Config for Runtime {
 	type Fungibles = Assets;
 	type OnChargeAssetTransaction = pallet_asset_tx_payment::FungiblesAdapter<
 		pallet_assets::BalanceToAssetBalance<Balances, Runtime, ConvertInto>,
 		AssetsToBlockAuthor<Runtime>,
 	>;
+	type WeightInfo = ();
```
Comment (vimukthi-git) [inline parachains/runtimes/assets/statemint/src/lib.rs:544]: is this supposed to be without weights?

File: parachains/runtimes/assets/statemint/src/lib.rs [hunk 2]
```diff
@@ -531,12 +531,23 @@ impl pallet_collator_selection::Config for Runtime {
 	type WeightInfo = weights::pallet_collator_selection::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub const UseUserConfiguration: bool = false;
```
Comment (vimukthi-git) [inline parachains/runtimes/assets/statemint/src/lib.rs:535]: What does this do exactly?

**Review by vimukthi-git — APPROVED** [review]

File: parachains/runtimes/assets/statemint/src/lib.rs (hunk 2 above)
Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:535]: you can check this PR https://github.com/paritytech/substrate/pull/12087/files#diff-4b420db30e4fc16308ea5d9858a9e56e1895d670f903a76fde733e316125b2a1

File: parachains/runtimes/assets/statemint/src/lib.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:544]: we do not use statemine. we do not host statemine. 
so will generate weights only in our runtime.

Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:544]: we do not host in node, but use in rust xcmp tests as some parachain, so it should be compilable. but do not care wights in such tests.

File: parachains/runtimes/assets/statemint/src/lib.rs (hunk 2 above)
Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:535]: also please check this diagram. https://github.com/ComposableFi/composable/blob/main/docs/runtime-upgrades-and-versioning.md updates are done according compile time dependencies .

File: parachains/runtimes/assets/statemint/src/lib.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:544]: so for sure parity will not host it without weighs, but we do not host that stuff.

Comment (dzmitry-lahoda) [reply parachains/runtimes/assets/statemint/src/lib.rs:544]: also please check official guide from parity on PR. you have to create satellite prs to make their tests path. so really this change to great extend to pass their CI.

**Comment by dzmitry-lahoda** [comment]
current status from parity is 
image

**Review by KaiserKarel — APPROVED** [review]

**Pull request merged**